parts/linux/cloud-init/artifacts/kubelet.service

```diff
@@ -20,6 +20,8 @@ ExecStartPre=/bin/mount --make-shared /var/lib/kubelet
 ExecStartPre=-/sbin/ebtables -t nat --list
 ExecStartPre=-/sbin/iptables -t nat --numeric --list
 
+ExecStartPre=/bin/bash /opt/azure/containers/validate-kubelet-credentials.sh
```

Comment:
are `ExecStartPre` blocking and sequential ? since this is expecting that `/var/lib/kubelet` is ready and we are mount that folder in a `ExecStartPre` right before.

Reply:
yes -

`Additional commands that are executed before or after the command in ExecStart=, respectively. Syntax is the same as for ExecStart=. Multiple command lines are allowed, regardless of the service type (i.e. Type=), and the commands are executed one after the other, serially.`

https://www.freedesktop.org/software/systemd/man/latest/systemd.service.html#ExecStartPre=